apps/sms/views/shared/js/utils.js

```diff
@@ -945,24 +945,33 @@
     },
 
     /**
-     * Returns Promise instance that is resolved instantly if document is
-     * visible (document.hidden === false), othewise it's resolved only once
-     * `visibilitychange` event is fired.
-     * @returns {Promise.<void>}
+     * Returns Promise instance that is resolved only once specified event is
+     * fired on the specified event target.
+     * @param {EventTarget} eventTarget Event target to listen event on.
+     * @param {String} eventName Name of the event to listen for.
+     * @returns {Promise.<*>} Promise resolved with event object if any.
      */
-    onceDocumentIsVisible() {
-      if (!document.hidden) {
-        return Promise.resolve();
-      }
-
+    onceEventIsFired(eventTarget, eventName) {
```

Comment:
I mean case like this:
```
console.log(document.readyState === 'complete') // 'true'
Utils.onceEventIsFired(window, 'load');
```

In this case ```load``` event will never fire and we'll have event handler forever. So that's the main reason I don't feel 100% strong about this method, either in its description should be said that user will have to make sure that this event will fire eventually. So that it should always be:
```
if (document.readyState !== 'complete') {
  Utils.onceEventIsFired(window, 'load');
}
```

But we don't have ```document.readyState```-like api for the navigation transition events and it's not clear if it's implemented at all :)